tools/slicec-cs/src/decoding.rs

```diff
@@ -397,6 +415,42 @@ fn decode_stream_parameter(type_ref: &TypeRef, namespace: &str, encoding: Encodi
     }
 }
 
+fn decode_result_field(type_ref: &TypeRef, namespace: &str, encoding: Encoding) -> CodeBlock {
+    if type_ref.is_optional {
+        let mut decode_func_body = decode_func_body(type_ref, namespace, encoding);
+
+        // TODO: it's lame to have to do this here. We should provide a better API.
+        if matches!(type_ref.concrete_type(), Types::Sequence(_) | Types::Dictionary(_)) {
+            write!(
+                decode_func_body,
+                " as {}",
+                type_ref.cs_type_string(namespace, TypeContext::Nested, false),
+            );
+        }
+
+        decode_func_body.indent();
+
+        // TODO: this won't result is a nice looking line when decode_func_body is multi-line
+        CodeBlock::from(format!(
+            "(ref SliceDecoder decoder) => decoder.DecodeBool() ? {decode_func_body} : null"
+        ))
+    } else {
+        let mut decode_func = decode_func(type_ref, namespace, encoding);
+
+        // Ditto
+        if matches!(type_ref.concrete_type(), Types::Sequence(_) | Types::Dictionary(_)) {
+            write!(
+                decode_func,
+                " as {}",
+                type_ref.cs_type_string(namespace, TypeContext::Nested, false),
+            );
+        }
+
+        decode_func.indent(); // TODO: this doesn't work
```

Comment:
@externl, do you remember how the indentation works?

Reply:
Ident returns a new indexed code block, it does not modify the existing one.

Reply:
``` 
pub fn indent(&mut self) -> &mut Self {
    self.content = self.content.replace('\n', "\n    ");
    self
}

Reply:
We never indent the first line. I suspect this is the issue. I guess we could name it better. The logic for this is to work in the more common usage (at least in our code so far).
```rust
format!("
foo(
    {bar},
    b,
    c
)
", bar.indent()); 
```